### ISSUE TITLE
`virtualisation.virtualbox.host.package` should check for kernel version

### ISSUE THREAD

**Issue by pca006132** [NixOS/nixpkgs#245834]
### Describe the bug

TLDR: Virtualbox may not support the latest kernel version.

I am currently using 23.05 release with `xanmod_latest` kernel, which is kernel 6.4.1, and the virtualbox package in the 23.05 release is 7.0.8 which does not support kernel 6.4. Changing the virtualbox package to the one in unstable channel (7.0.10) fixes the problem.

I think in general we should check for kernel version and warn the user about incompatibility.

### Steps To Reproduce
1. Use kernel 6.4 and enable virtualbox in 23.05 release.

The vm is stuck in starting state, and cannot be stopped.

### Expected behavior
Warning that virtualbox will/may not work correctly with this kernel version.

### Notify maintainers

@svanderburg 

Not sure if I should notify people who wrote the nixos module, as the git blames are really old.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.1-xanmod1, NixOS, 23.05 (Stoat), 23.05.20230718.08700de`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos-22.11, nixpkgs-unstable"`
 - channels(pca006132): `""`
 - nixpkgs: `/etc/nix/channels/nixpkgs`
```
